backend/test/classes.controller.e2e-spec.ts

```diff
@@ -0,0 +1,112 @@
+import { INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { getApp } from "./helper";
+import { classes, defaultAdmin, defaultTeacher, users } from "test/seed";
+
+const checkClassesInList = (expectedClasses, returnedClasses): void => {
+  expect(returnedClasses).toHaveLength(expectedClasses.length);
+  returnedClasses.forEach((returnedClass) => {
+    const { teacherId, ...referenceClass } = expectedClasses.find(
+      (c) => c.id == returnedClass.id,
+    );
+    expect(returnedClass).toMatchObject(referenceClass);
+
+    const {
+      type: _ignore_type,
+      email: _ignore_email,
```

Comment:
do we want to assert the absence of those fields instead? (also fine for me if we add a comment and do it later)

Reply:
they're there and I literally don't want them in the `referenceTeacher` :-)

Reply:
But longer term, I'd like to have proper matchers for the various DTOs.

Reply:
ah I see ^^